src/main/java/io/appform/dropwizard/actors/utils/CommonUtils.java

```diff
@@ -44,4 +46,22 @@ public static boolean isRetriable(Set<String> retriableExceptions, Throwable exc
                 || (null != exception
                 && retriableExceptions.contains(exception.getClass().getSimpleName()));
     }
+
+    public static <T> Optional<T> extractMessagePropertiesHeader(final Map<String, Object> headers,
+                                                          final String key,
+                                                          final Class<T> clazz) {
+        if (null == headers || !headers.containsKey(key)) {
```

Comment:
containsKey and get incur the same cost. Do a get() and check directly if it is null or not if headers is not null. Basically rewrite this function for efficiency